packages/trader/src/App/Containers/RealAccountSignup/personal-details.jsx

```diff
@@ -0,0 +1,268 @@
+            should_show_calendar: false,
+        });
+    }
+
+    componentWillUnmount() {
+        document.removeEventListener('mousedown', this.handleClick);
+    }
+
+    handleFocus = () => {
+        this.setState({
+            should_show_calendar: true,
+        });
+    };
+
+    render() {
+        return (
+            <Field
+                id={this.props.id}
+                name={this.props.name}
+                render={({
+                    field: { name, value },
+                    id,
+                    label,
+                    className,
+                    form : { setFieldValue },
+                }) => (
+                    <div className='datepicker'>
+                        <Input
+                            name={name}
+                            className={classNames(className, 'dc-input--no-placeholder')}
+                            id={id}
+                            label={label}
+                            placeholder={this.props.label}
+                            onFocus={this.handleFocus}
+                            onBlur={this.handleBlur}
+                            value={value ? toMoment(value).format('YYYY-MM-DD') : ''}
+                            readOnly
+                        />
+                        <CSSTransition
+                            in={this.state.should_show_calendar}
+                            timeout={100}
+                            classNames={{
+                                enter    : 'datepicker__picker--enter datepicker__picker--bottom-enter',
+                                enterDone: 'datepicker__picker--enter-done datepicker__picker--bottom-enter-done',
+                                exit     : 'datepicker__picker--exit datepicker__picker--bottom-exit',
+                            }}
+                            unmountOnExit
+                        >
+                            <div
+                                className='datepicker__picker'
+                                ref={this.reference}
+                            >
+                                <DatePickerCalendar
+                                    max_date={this.state.max_date}
+                                    min_date={this.state.min_date}
+                                    date={this.state.date}
+                                    onChange={(val, type) => {
+                                        setFieldValue(name, val, true);
+                                        if (type === 'day') {
+                                            this.closeDatePicker();
+                                        }
+                                    }}
+                                    value={value}
+                                />
+                            </div>
+                        </CSSTransition>
+                    </div>
+                )}
+            />
+        );
+    }
+}
+
+const Input = (props) => {
+    const ref = React.createRef();
```

Comment:
is there a reason to create Input component for this? since we already have `Input` component in deriv-component

Reply:
I had problems with input component incorporating it with the Formik, it is fixed since then, but I haven't had time to refactor it. I'll add an item to improvements, and add a todo here. thanks :pray: